tests/java/lang/ssl_hostname_verifier/testdata/main.java

```diff
@@ -0,0 +1,124 @@
+import javax.net.ssl.X509TrustManager;
+import org.apache.http.conn.ssl.SSLSocketFactory;
+
+SSLSocketFactory socketFactory = SSLSocketFactory.getSocketFactory();
+
+// bearer:expected java_lang_ssl_hostname_verifier
+HostnameVerifier hostnameVerifier = org.apache.http.conn.ssl.SSLSocketFactory.ALLOW_ALL_HOSTNAME_VERIFIER;
+
+// bearer:expected java_lang_ssl_hostname_verifier
+HttpsURLConnection.setDefaultHostnameVerifier(hostnameVerifier);
+
+// bearer:expected java_lang_ssl_hostname_verifier
+HttpsURLConnection.setDefaultHostnameVerifier(NoopHostnameVerifier.INSTANCE);
+
+// bearer:expected java_lang_ssl_hostname_verifier
+socketFactory.setHostnameVerifier((X509HostnameVerifier) hostnameVerifier);
```

Comment:
@cfabianski I'm stuck on this last one (it is not caught): 

```java
HostnameVerifier hostnameVerifier = org.apache.http.conn.ssl.SSLSocketFactory.ALLOW_ALL_HOSTNAME_VERIFIER;
socketFactory.setHostnameVerifier((X509HostnameVerifier) hostnameVerifier);
```

I thought this pattern would catch it: https://github.com/Bearer/bearer-rules/pull/231/files#diff-eb911c5574ab45fb88bb10d073948c07043079e223275a36fc4b93b74adb4eb3R130-R137